keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -225,9 +226,9 @@ class SqlDeliveryConfigRepository(
           constraints = mapper.readValue(constraintsJson),
           notifications = mapper.readValue(notificationsJson ?: "[]")
         )
-      }
+      } ?: throw OrphanedResourceException(resourceId)
```

Comment:
It's no longer valid to not have a delivery config for a resource, hence the new exception throwing.